src/main/webapp/creatives/maps/simid_map_creative.test.js

```diff
@@ -181,36 +203,52 @@ test('testing that rejection for JSON parsing errors is working', () => {
         type: "SIMID:Player:init",
     };
     testMap.onInit(eventData);
-    
-    const instance = SimidProtocol.mock.instances[0];
-    const rejectMessageObject = instance.reject.mock.calls[0][1];
+
+    expect(mockReject.mock.calls.length).toBe(1);
+    const rejectMessageObject = mockReject.mock.calls[0][1];
     expect(rejectMessageObject.message).toBe("Invalid JSON input for ad parameters");
 });
 
-test('instance of map is instantiated on button click', () => {
+test('instance of map is instantiated if ad paused', async () => {
     const eventData = createInitData();
     testMap.onInit(eventData);
     testMap.onStart(startData);
+
     const findNearestButton = document.getElementById('findNearest');
     findNearestButton.dispatchEvent(new Event('click'));
+
+    await drivePromisesToCompletion();
```

Comment:
was the problem from yesterday b/c the promises were not fully executed before the expect??
Good job figuring that out! That was definitely a hard one to notice

Reply:
^^

Reply:
that was the issue for fixing kristen's tests, the issue for fixing the other tests was that the request functions had to be mocked by themselves and not as part of the simidProtocol mock. Then using this mocked function, I had to call mockReset so that the mock restarted in every test